src/python/tests/test_configuration_manager.py

```diff
@@ -0,0 +1,710 @@
+        """
+        stmt = sa.select([_configuration_tbl.c.value]).select_from(_configuration_tbl).where(
+            _configuration_tbl.c.key == 'boolean')
+        await self.setup_method()
+        await set_category_item_value_entry(category_name='boolean',
+                                            item_name='info', new_value_entry='True')
+        try:
+            async with aiopg.sa.create_engine(_CONNECTION_STRING) as engine:
+                async with engine.acquire() as conn:
+                    async for result in conn.execute(stmt):
+                        assert result is None
+        except Exception:
+            raise
+        await self.teardown_method()
+
+    async def test_get_category_item_value_entry(self):
+        """
+        Test that get_category_item_value_entry works properly
+        :Assert:
+            1. category_value.value gets returned and matches default
+            2. When updating value, the data retrieved for value gets updated
+        """
+        await self.setup_method()
+        await create_category(category_name='boolean', category_description='boolean type',
+                              category_value={'info': {
+                                  'description': 'boolean type with default False',
+                                  'type': 'boolean',
+                                  'default': 'False'}})
+        result = await get_category_item_value_entry(category_name='boolean', item_name='info')
+        assert result == 'False'
+
+        await set_category_item_value_entry(category_name='boolean',
+                                            item_name='info', new_value_entry='True')
+        result = await get_category_item_value_entry(category_name='boolean', item_name='info')
+        assert result == 'True'
+        await self.teardown_method()
+
+    async def test_get_category_item_value_entry_empty(self):
+        """
+        Test that None gets returned when either category_name and/or item_name don't exist
+        :assert:
+            1. Assert None is returned when item_name does not exist
+            2. Assert None is returned when category_name does not exist
+        """
+        await self.setup_method()
+        await create_category(category_name='boolean', category_description='boolean type',
+                              category_value={
+                                  'info': {
+                                      'description': 'boolean type with default False',
+                                      'type': 'boolean',
+                                      'default': 'False'}
+                              })
+        result = await get_category_item_value_entry(category_name='boolean', item_name='data')
+        assert result is None
+
+        result = await get_category_item_value_entry(category_name='integer', item_name='info')
+        assert result is None
+        await self.teardown_method()
+
+    async def test_get_category_item(self):
```

Comment:
seems its already tested; we can not keep mixing scenarios

Reply:
meant to test get_all_category_names. this has been fixed, and simplified to only test a few categories.